frontend/src/app/event/event-filter/event-filter.pipe.ts

```diff
@@ -45,3 +45,18 @@ export class EventFilterPipe implements PipeTransform {
     }
   }
 }
+
+@Pipe({
+  name: 'eventUserRegistrationStatusPipe'
+})
+export class EventUserRegistrationStatusPipe implements PipeTransform {
+  transform(statuses: UserRegistrationStatus[], event_id: number): boolean {
+    let filteredStatuses = statuses.filter((status) => {
+      return status.event_id == event_id;
+    });
+
+    return filteredStatuses.length > 0
+      ? filteredStatuses[0].is_registered
+      : false;
+  }
```

Comment:
Seeing this complexity carried forward to the front-end, I am increasingly feeling like doing more heavy lifting and specificity in the backend (e.g. having an endpoint that returns event models that are subclasses of event with user-specific registration field added) will make the front-end implementation require less gymnastics.